### ISSUE TITLE
Nested $refs are not resolved properly if used in "allOf" elsewhere

### ISSUE THREAD

**Issue by ringgelerch** [swagger-api/swagger-ui#1713]
In my example spec (you will find it as the end of this description) the reference to "$ref: '#/definitions/2ndLvlRef'" is not resolved properly. As you can see from the first snapshot the property "notDisplayed" is not visible.

![fail](https://cloud.githubusercontent.com/assets/2796210/10840749/b032799c-7ee2-11e5-8905-4d6234482dfb.PNG)

When I comment out the definition of "OtherDef" (this one isn't used anywhere in my stripped example!) or remove the "allOf" or replace it with an usual property reference everthing is fine as you can see here:

![ok](https://cloud.githubusercontent.com/assets/2796210/10840807/537283b8-7ee3-11e5-935d-2c7c5fcf3cad.PNG)

Example spec to reproduce the failure:

```
swagger: '2.0'
info:
  title: TEST
  version: V1
host: localhost
schemes:
- http
basePath: /v1
produces:
- application/json

paths:
  /test:
    get:
      summary: Test
      responses:
        200:
          schema:
            $ref: '#/definitions/TestDef'

definitions:

  TestDef:
    properties:
      test:
        $ref: '#/definitions/1stLvlRef'

  OtherDef:
    allOf:
    - $ref: "#/definitions/1stLvlRef"

  1stLvlRef:
    properties:
      dummy:
        type: string
      notDisplayed:
        $ref: '#/definitions/2ndLvlRef'

  2ndLvlRef:
    properties:
      dummy:
        type: string
```

Did I miss something or is this a bug of the resolver. It would be greate if you could offer a solution soon.

**Comment by hassansin** [comment]
I'm having similar issue. But in my case, the problem is in the base model, not in the extended model: #1723

**Comment by webron** [comment]
This should be resolved in 3.0. Please reopen if you experience otherwise.
